scripts/build/builders/nuttx.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+from enum import Enum, auto
+
+from .gn import Builder
+
+
+class NuttXApp(Enum):
+    LIGHT = auto()
+
+    def ExampleName(self):
+        if self == NuttXApp.LIGHT:
+            return 'lighting-app'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+    def AppNamePrefix(self, chip_name):
+        if self == NuttXApp.LIGHT:
+            return ('chip-%s-lighting-example' % chip_name)
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+
+class NuttXBoard(Enum):
+    SIM = auto()
+
+
+def NuttXTarget(board, app):
+    if board == NuttXBoard.SIM:
+        if app == NuttXApp.LIGHT:
+            return 'sim:matter'
+
+
+class NuttXBuilder(Builder):
+
+    def __init__(self,
+                 root,
+                 runner,
+                 app: NuttXApp = NuttXApp.LIGHT,
+                 board: NuttXBoard = NuttXBoard.SIM,
+                 enable_rpcs: bool = False,
+                 module_type: str = 'sim',
+                 baudrate=2000000,
+                 enable_shell: bool = False,
+                 enable_cdc: bool = False,
+                 enable_resetCnt: bool = False,
+                 enable_rotating_device_id: bool = False,
+                 enable_ethernet: bool = False,
+                 enable_wifi: bool = False,
+                 enable_thread: bool = False,
+                 enable_frame_ptr: bool = False,
+                 enable_heap_monitoring: bool = False,
+                 use_matter_openthread: bool = False,
+                 ):
+
+        nuttx_chip = 'nuttx'
+
+        super(NuttXBuilder, self).__init__(
+            root=os.path.join(root, 'examples',
+                              app.ExampleName(), nuttx_chip),
+            runner=runner
+        )
+
+        self.argsOpt = []
+        self.chip_name = nuttx_chip
+        self.enable_frame_ptr = enable_frame_ptr
+        self.app = app
+        self.board = board
+        # self.nuttx_dir = os.path.join(self.output_dir, 'nuttx')
+        # self.nuttx_apps_dir = os.path.join(self.output_dir, 'apps')
```

Comment:
do not check in commented out code unless there is some future explanation about what this is supposed to do

Reply:
Done.